### PR TITLE
Webitools 161 use fast scan URL instead of discover rediscover application lfo

### PR DESCRIPTION
**Pull request by leon2Meudon92**

### CODE REVIEW

File: aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/JnksLogPollingProviderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        this.verbose = verbose;
+        this.jobsService = jobsService;
+        this.sleepDuration = sleepDuration;
+    }
+
+    @Override
+    public String pollJobLog(String jobGuid) throws JobServiceException {
+        JobExecutionDto jobExecutionDto = jobsService.pollAndWaitForJobFinished(jobGuid,
+                this::callbackFunction, getPollingCallback(log), Function.identity(), () -> TimeUnit.SECONDS.toMillis(sleepDuration));
+        //s -> s.getState() == JobState.COMPLETED ? s : null);
```
Comment (AntoineCarasso) [inline aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/JnksLogPollingProviderServiceImpl.java:44]: Shouldn't this commented code should be removed ?

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java [hunk 2]
```diff
@@ -38,11 +38,16 @@ public class OnboardApplicationDeepAnalysisCommand extends BasicCollable {
             description = "The name of the snapshot to create")
     private String snapshotName;
 
+    @CommandLine.Option(names = {"--sleep-duration"},
+            description = "Amount of time  used to fetch the ongoing job status (specified in seconds ). The default value is: ${DEFAULT-VALUE}",
```
Comment (AntoineCarasso) [inline aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java:42]: I would rephrase this as "Number of seconds ..." to make this message shorter, and avoid the word "amount" that is often used when speaking of a quantity of money.

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationFastScanCommand.java [hunk 3]
```diff
@@ -61,11 +61,16 @@ public class OnboardApplicationFastScanCommand extends BasicCollable {
             , description = "Project's exclusion rules, separated with comma. Valid values: ${COMPLETION-CANDIDATES}")
     private ExclusionRuleType[] exclusionRules;
 
+    @CommandLine.Option(names = {"--sleep-duration"},
+            description = "Amount of time  used to fetch the ongoing job status (specified in seconds ). The default value is: ${DEFAULT-VALUE}",
```
Comment (AntoineCarasso) [inline aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationFastScanCommand.java:65]: Same comment as before: I would rephrase this as "Number of seconds ..."

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/providers/CliLogPollingProviderImpl.java [hunk 4]
```diff
@@ -1,23 +1,42 @@
...
     }
 
     @Override
     public String pollJobLog(String jobGuid) throws JobServiceException {
+        return jobsService.pollAndWaitForJobFinished(jobGuid,
+                jobStep -> log.info("Current step is : " + jobStep.getCurrentStep()),
+                !verbose ? null : this::printLog,
+                (s) -> s.getState() == JobState.COMPLETED ? s.getJobParameters().get("appGuid") : null,
+                () -> TimeUnit.SECONDS.toMillis(sleepDuration));
+        /*
```
Comment (AntoineCarasso) [inline aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/providers/CliLogPollingProviderImpl.java:32]: Commented code

File: aip-console-jenkins/README.md [hunk 5]
```diff
@@ -341,6 +341,7 @@ Under `Advanced Settings` you will find the following parameters :
 * *Application Name*: Application on which to perform Deep Analysis.
 * *Snapshot Name*: Name of the snapshot. CAST Imaging Console will automatically assign one to the application if this
   field is empty.
+* *sleep duration* : Amount of seconds used to fetch the ongoing job status (defaulted to **15s**).
```
Comment (AntoineCarasso) [inline aip-console-jenkins/README.md:344]: I would replace "Amount" by "Number" because "amount" is related to a quantity of money

File: aip-console-jenkins/README.md [hunk 6]
```diff
@@ -359,6 +360,7 @@ Under `Advanced Settings` you will find the following parameters :
     replaced by the current jobs' workspace path.
 * *Exclusion patterns* : List of file patterns to exclude in the delivery, separated by a comma. Each patterns needs to
   follow the syntax of [glob patterns](https://www.malikbrowne.com/blog/a-beginners-guide-glob-patterns)
+* *sleep duration* : Amount of seconds used to fetch the ongoing job status (defaulted to **1s**).
```
Comment (AntoineCarasso) [inline aip-console-jenkins/README.md:363]: I would replace "Amount" by "Number" because "amount" is related to a quantity of money

File: aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/OnboardApplicationDeepAnalysisBuilder.java [hunk 7]
```diff
@@ -142,7 +101,7 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
             String caipVersion = app.getCaipVersion();
             String targetNode = app.getTargetNode();
             boolean verbose = getDescriptor().configuration.isVerbose();
-            OnboardApplicationDeepAnalysisBuilder.JnksLogPollingProviderImpl jnksLogPollingProvider = new JnksLogPollingProviderImpl(run, listener, verbose);
+            JnksLogPollingProviderServiceImpl jnksLogPollingProvider = new JnksLogPollingProviderServiceImpl(jobsService, run, listener, verbose, getSleepDuration());
```
Comment (AntoineCarasso) [inline aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/OnboardApplicationDeepAnalysisBuilder.java:104]: If "Jnks" stands for "Jenkins", I would write it "Jenkins" for class & member names

File: aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/JnksLogPollingProviderServiceImpl.java (hunk 1 above)
Comment (leon2Meudon92) [reply aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/JnksLogPollingProviderServiceImpl.java:44]: hmm!

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java (hunk 2 above)
Comment (leon2Meudon92) [reply aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java:42]: let go for number of seconds

File: aip-console-jenkins/README.md (hunk 5 above)
Comment (leon2Meudon92) [reply aip-console-jenkins/README.md:344]: done with the earlier remark
